src/ipc/handler.c

```diff
@@ -803,16 +856,27 @@ static int ipc_comp_value(uint32_t header, uint32_t cmd)
 	}
 	
 	/* get component values */
-	ret = ipc_comp_cmd(comp_dev->cd, cmd, &data);
+	ret = ipc_comp_cmd(comp_dev->cd, cmd, _data, SOF_IPC_MSG_MAX_SIZE);
```

Comment:
Why are we sending Max size as a parameter ?  Do we even have to since we can relay on
```
struct sof_ipc_cmd_hdr {
	uint32_t size;			/**< size of structure */
	uint32_t cmd;			/**< SOF_IPC_GLB_ + cmd */
} __attribute__((packed));
```

Reply:
Why do we even need copied data  since we use original _data to work on and reply?

Reply:
The usage for get()/set() is that components can also modify/extend data and return it to the host.